neo/rawio/neuralynxrawio.py

```diff
@@ -440,46 +440,67 @@ def read_ncs_files(self, ncs_filenames):
         data0 = np.memmap(filename0, dtype=ncs_dtype, mode='r', offset=HEADER_SIZE)
 
         gap_indexes = None
+        lost_indexes = None
+
         if self.use_cache:
             gap_indexes = self._cache.get('gap_indexes')
+            lost_indexes =  self._cache.get('lost_indexes')
 
         # detect gaps on first file
-        if gap_indexes is None:
+        if (gap_indexes is None) or (lost_indexes is None):
+
             # this can be long!!!!
             timestamps0 = data0['timestamp']
             deltas0 = np.diff(timestamps0)
 
             # It should be that:
             # gap_indexes, = np.nonzero(deltas0!=good_delta)
-            # but for a file I have found many deltas0==15999 deltas0==16000
+            # but for a file I have found many deltas0==15999, 16000, 16001 (for sampling at 32000)
             # I guess this is a round problem
             # So this is the same with a tolerance of 1 or 2 ticks
-            mask = deltas0 != good_delta
-            for tolerance in (1, 2):
-                mask &= (deltas0 != good_delta - tolerance)
-                mask &= (deltas0 != good_delta + tolerance)
+            max_tolerance = 2
+            mask = np.abs((deltas0 - good_delta).astype('int64'))>max_tolerance
+
```

Comment:
More easy to read. The previous implementation was a bit faster in my memory but no sure.
In any case, it is better to have something readable.

Reply:
> More easy to read. The previous implementation was a bit faster in my memory but no sure.
> In any case, it is better to have something readable.

@samuelgarcia  do you mean, I should've left the original version?

Reply:
No I mean it is perfect because more readable.